### PR TITLE
8288599: com/sun/management/OperatingSystemMXBean/TestTotalSwap.java: Expected total swap size ... but getTotalSwapSpaceSize returned ...

### PR DESCRIPTION
**Pull request by lmesnik**
The failure is very intermittent and I wasn't able to reproduce it.

I suspect that the swap might be changed during test execution so I add the corresponding check. 

If test still fails after fix we should at least exclude this potential problem.

---------
### Progress
- [x] Change must be properly reviewed (1 review required, with at least 1 [Reviewer](https://openjdk.org/bylaws#reviewer))
- [x] Change must not contain extraneous whitespace
- [x] Commit message must refer to an issue

### Issue
 * [JDK-8288599](https://bugs.openjdk.org/browse/JDK-8288599): com/sun/management/OperatingSystemMXBean/TestTotalSwap.java: Expected total swap size ... but getTotalSwapSpaceSize returned ...

### Reviewers
 * [Serguei Spitsyn](https://openjdk.org/census#sspitsyn) (@sspitsyn - **Reviewer**)
 * [Kevin Walls](https://openjdk.org/census#kevinw) (@kevinjwalls - Committer)

### Reviewing
<details><summary>Using <code>git</code></summary>

Checkout this PR locally: \
`$ git fetch https://git.openjdk.org/jdk pull/9174/head:pull/9174` \
`$ git checkout pull/9174`

Update a local copy of the PR: \
`$ git checkout pull/9174` \
`$ git pull https://git.openjdk.org/jdk pull/9174/head`

</details>
<details><summary>Using Skara CLI tools</summary>

Checkout this PR locally: \
`$ git pr checkout 9174`

View PR using the GUI difftool: \
`$ git pr show -t 9174`

</details>
<details><summary>Using diff file</summary>

Download this PR as a diff file: \
<a href="https://git.openjdk.org/jdk/pull/9174.diff">https://git.openjdk.org/jdk/pull/9174.diff</a>

</details>

### CODE REVIEW

**Review by sspitsyn — APPROVED** [review]
Looks good to me.
Thanks,
Serguei

File: test/jdk/com/sun/management/OperatingSystemMXBean/TestTotalSwap.java [hunk 1]
```diff
@@ -71,19 +57,31 @@ public class TestTotalSwap {
...
 
+        long expected_swap_size = getSwapSizeFromOs();
         long size = mbean.getTotalSwapSpaceSize();
 
-        System.out.println("Total swap space size in bytes: " + size);
+        System.out.println("Total swap space size from OS in bytes: " + expected_swap_size);
+        System.out.println("Total swap space size in MBean bytes: " + size);
+
+        // if swap data from OS chnaged re-read OS and MBean data
+        while (expected_swap_size != getSwapSizeFromOs()) {
```
Comment (kevinjwalls) [inline test/jdk/com/sun/management/OperatingSystemMXBean/TestTotalSwap.java:76]: Can we put a limit on this?
OS swap usage could be continually changing, this seems worth a try but we want to escape and give up after some number of tries?
If there is some variation that's making us fail, we could also mask off some number of low bits, or a byte or more even, to do an approximate comparison?  Might want to try that if retrying does not succeed.

**Review by kevinjwalls — APPROVED** [review]

File: test/jdk/com/sun/management/OperatingSystemMXBean/TestTotalSwap.java (hunk 1 above)
Comment (lmesnik) [reply test/jdk/com/sun/management/OperatingSystemMXBean/TestTotalSwap.java:76]: I doubt that swap might change so often. The failure is reproduced once in several months.
Let's check the logs if the test failed.

**Pull request closed without merging**